apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php

```diff
@@ -232,32 +234,45 @@ function load_wpcom_block_editor_nux() {
 add_action( 'plugins_loaded', __NAMESPACE__ . '\load_wpcom_block_editor_nux' );
 
 /**
- * Load editing toolkit block patterns from the API
+ * Return a function that loads and register block patterns from the API. This
+ * function can be registered to the `rest_dispatch_request` filter.
  *
- * @param obj $current_screen The current screen object.
+ * @param Function $register_patterns_func A function that when called will
+ * register the relevant block patterns in the registry.
  */
-function load_block_patterns_from_api( $current_screen ) {
-	if ( ! apply_filters( 'a8c_enable_block_patterns_api', false ) ) {
-		return;
-	}
-
-	$is_site_editor = ( function_exists( 'gutenberg_is_edit_site_page' ) && gutenberg_is_edit_site_page( $current_screen->id ) );
-
-	if ( ! $current_screen->is_block_editor && ! $is_site_editor ) {
-		return;
-	}
+function load_block_patterns_from_api( $register_patterns_func ) {
+	/**
+	 * Load editing toolkit block patterns from the API.
+	 *
+	 * It will only register the patterns for certain allowed requests and
+	 * return early otherwise.
+	 *
+	 * @param mixed           $response
+	 * @param WP_REST_Request $request
+	 */
+	return function ( $response, $request ) use ( $register_patterns_func ) {
+		$route           = $request->get_route();
+		$request_allowed = preg_match( '/^\/wp\/v2\/sites\/[0-9]+\/block\-patterns\/(patterns|categories)$/', $route );
```

Comment:
Ah, I just realized this might not work for AT sites, as the route will be different, I think :/ we might need to adapt the regex here to work for it to work in AT sites.

Reply:
Yes, you'll need to match routes both with and without the `sites/<wpcom_site>` infix. I tried to find some way to avoid this, like looking at the 4th `$handler` parameter of the filter callback, but the "canonical" `/wp/v2/block-patterns/patterns` route is not preserved anywhere.

Reply:
I've updated the regex to optionally match the `sites/123/` portion of the string, and added phpunit test cases to make sure that both the `/wp/v2/block-patterns/` and `/wp/v2/sites/123/block-patterns` prefixes are allowed.